exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/NetworkFunctions.java

```diff
@@ -0,0 +1,668 @@
+      out.end = outputValue.getBytes().length;
+      buffer.setBytes(0, outputValue.getBytes());
+    }
+
+
+  }
+
+  /**
+   * This function returns true if a given IPv4 address is private, false if not.
+   */
+
+  @FunctionTemplate(
+    name = "is_private_ip",
+    scope = FunctionTemplate.FunctionScope.SIMPLE,
+    nulls = FunctionTemplate.NullHandling.NULL_IF_NULL
+  )
+  public static class IsPrivateIP implements DrillSimpleFunc {
+
+    @Param
+    VarCharHolder inputTextA;
+
+    @Output BitHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+
+    public void setup() {
+    }
+
+
+    public void eval() {
+      String ipString = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(inputTextA.start, inputTextA.end, inputTextA.buffer);
+
+      String[] ipAddressInArray = ipString.split("\\.");
+
+      int result = 0;
+
+      int[] octets = new int[3];
+
+      for( int i = 0; i < 3; i++ ){
+        octets[i] = Integer.parseInt( ipAddressInArray[i] );
+        if( octets[i] > 255 || octets[i] < 0 ) {
+          result = 0;
+        }
+      }
+
+      if( octets[0] == 192 && octets[1] == 168 ) {
+        result = 1;
+      }
+      else if (octets[0] == 172 && octets [1] >= 16 && octets[1] <= 31 ){
+        result = 1;
+      }
+      else if( octets[0] == 10 ) {
+        result = 1;
+      }
+      else {
+        result = 0;
+      }
+
+      out.value = result;
+    }
+  }
+  /**
+   *  This function converts an IPv4 address into a BigInt.  Useful for sorting IPs, or looking for IP ranges.
+   *  IE:
+   *  SELECT *
+   *  FROM <data>
+   *  ORDER BY inet_aton( ip ) ASC
+   */
+  @FunctionTemplate(
+    name = "inet_aton",
+    scope = FunctionTemplate.FunctionScope.SIMPLE,
+    nulls = FunctionTemplate.NullHandling.NULL_IF_NULL
+  )
+  public static class InetAtonFunction implements DrillSimpleFunc {
+
+    @Param
+    VarCharHolder inputTextA;
+
+    @Output BigIntHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+
+    public void setup() {
+    }
+
+
+    public void eval() {
+      String ipString = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(inputTextA.start, inputTextA.end, inputTextA.buffer);
+      if( ipString == null || ipString.isEmpty() || ipString.length() == 0 ){
+        out.value = 0;
+      } else {
+        String[] ipAddressInArray = ipString.split("\\.");
+
+        long result = 0;
+        for (int i = 0; i < ipAddressInArray.length; i++) {
+          int power = 3 - i;
+          int ip = Integer.parseInt(ipAddressInArray[i]);
+          result += ip * Math.pow(256, power);
+        }
+
+        out.value = result;
+      }
+    }
+  }
+
+  /**
+   * Returns true if the input string is a valid IP address
+   */
+  @FunctionTemplate(
+    name = "is_valid_IP",
+    scope = FunctionTemplate.FunctionScope.SIMPLE,
+    nulls = FunctionTemplate.NullHandling.NULL_IF_NULL
+  )
+  public static class IsValidIPFunction implements DrillSimpleFunc {
```

Comment:
That was my original plan, however the commons validator has a dependency on commons-logging which is banned in Drill.  I wasn't sure why that's the case, but if we can fix that somehow, I can reinstate the code using commons-validator.

Reply:
Why it's banned in Drill? Can we exclude it, for example?

Reply:
I'm not really sure why it's banned.  I'd love to exclude it, but I couldn't seem to get Drill to build when I tried putting that exclusion in the java-exec pom file.  Any assistance would be greatly appreciated.

Reply:
Also you may ask on dev list regarding `commons-logging` usage on Drill dev list.

Reply:
We exclude `commons-logging` because Drill uses `logback` for logging.
Excluding it from `commons-validator` should work perfectly fine (I have tried).

```
    <dependency>
      <groupId>commons-validator</groupId>
      <artifactId>commons-validator</artifactId>
      <version>1.4.0</version>
      <exclusions>
      <exclusion>
        <groupId>commons-logging</groupId>
        <artifactId>commons-logging</artifactId>
      </exclusion>
      </exclusions>
    </dependency>
```

Reply:
Fixed